### PR TITLE
Refactor | Move the code into different functions

### PR DESCRIPTION
**Pull request by MatiasOlivera**
Issue #289

### CODE REVIEW

**Comment by MatiasOlivera** [comment]
@paulmelnikow Yes, there are other things that can be improved. I will review the tests

**Comment by ForsakenHarmony** [comment]
yeah, tests are unrelated I think, just stupidity on uglify's part

**Comment by ForsakenHarmony** [comment]
Feel free to update the snapshots

**Comment by MatiasOlivera** [comment]
I can't update the snapshots, because the tests are working in my machine 😱

Tests

```
λ node --version
v8.11.3
```
```
λ npm --version
6.1.0
```

**Comment by Andarist** [comment]
Maybe you have to clean npm cache and try to reinstall? If that won't help I'll look into it.

**Comment by ForsakenHarmony** [comment]
remove `node_modules` and `package-lock.json` and or try yarn

**Comment by MatiasOlivera** [comment]
I did what you said. For some reason, the tests are creating **new folders with the TS declaration files** (`index.d.ts`). These files already exist in `/dist`

New files

The snapshots are failing because it is **comparing the wrong directories**

Snapshots1
Snapshots2

The test result

```
 › 16 snapshots written.
 › 28 snapshots failed.
 › 30 snapshots obsolete.
   • fixtures custom-source 8
   • fixtures custom-source 9
   • fixtures custom-source 10
   • fixtures custom-source 11
   • fixtures custom-source 12
   • fixtures custom-source 13
   • fixtures custom-source 14
   • fixtures custom-source 15
   • fixtures custom-source 16
   • fixtures custom-source 17
   • fixtures custom-source 18
   • fixtures custom-source 19
   • fixtures custom-source 20
   • fixtures custom-source 21
   • fixtures custom-source 22
   • fixtures default-named 8
   • fixtures default-named 9
   • fixtures default-named 10
   • fixtures default-named 11
   • fixtures default-named 12
   • fixtures default-named 13
   • fixtures default-named 14
   • fixtures name-custom-amd 8
   • fixtures name-custom-amd 9
   • fixtures name-custom-amd 10
   • fixtures name-custom-amd 11
   • fixtures name-custom-amd 12
   • fixtures name-custom-amd 13
   • fixtures name-custom-amd 14
   • fixtures name-custom-amd 15
Snapshot Summary
 › 16 snapshots written from 1 test suite.
 › 28 snapshots failed from 1 test suite. Inspect your code changes or run `npm test -- -u` to update them.
 › 30 snapshots obsolete from 1 test suite. To remove them all, run `npm test -- -u`.
   ↳ test/index.test.js
       • fixtures custom-source 8
       • fixtures custom-source 9
       • fixtures custom-source 10
       • fixtures custom-source 11
       • fixtures custom-source 12
       • fixtures custom-source 13
       • fixtures custom-source 14
       • fixtures custom-source 15
       • fixtures custom-source 16
       • fixtures custom-source 17
       • fixtures custom-source 18
       • fixtures custom-source 19
       • fixtures custom-source 20
       • fixtures custom-source 21
       • fixtures custom-source 22
       • fixtures default-named 8
       • fixtures default-named 9
       • fixtures default-named 10
       • fixtures default-named 11
       • fixtures default-named 12
       • fixtures default-named 13
       • fixtures default-named 14
       • fixtures name-custom-amd 8
       • fixtures name-custom-amd 9
       • fixtures name-custom-amd 10
       • fixtures name-custom-amd 11
       • fixtures name-custom-amd 12
       • fixtures name-custom-amd 13
       • fixtures name-custom-amd 14
       • fixtures name-custom-amd 15

Test Suites: 1 failed, 1 total
Tests:       7 failed, 18 passed, 25 total
Snapshots:   28 failed, 30 obsolete, 16 written, 127 passed, 171 total
Time:        75.619s, estimated 107s
Ran all test suites.
```

File: src/index.js [hunk 1]
```diff
@@ -53,105 +48,65 @@ function formatSize(size, filename, type, raw) {
...
-	let main = resolve(cwd, options.output || options.pkg.main || 'dist');
-	if (!main.match(/\.[a-z]+$/) || (await isDir(main))) {
-		main = resolve(main, `${removeScope(options.pkg.name)}.js`);
-	}
-	options.output = main;
+	options.input = await getInput({
+		entries: options.entries,
+		cwd,
+		source: options.pkg.source,
+		module: options.pkg.module,
```
Comment (jonathantneal) [inline src/index.js]: I’m curious to learn, unless this is a mistake; how is `module` used as an input here?

**Pull request merged**